lib/datadog/opentelemetry/sdk/propagator.rb

```diff
@@ -41,8 +41,9 @@ def extract(
           digest = @datadog_propagator.extract(carrier)
           return context unless digest
 
-          trace_id = to_otel_id(digest.trace_id)
-          span_id = to_otel_id(digest.span_id)
+          # Converts the {Numeric} Datadog id object to OpenTelemetry's byte array format.
+          trace_id = [format('%032x', digest.trace_id)].pack('H32')
+          span_id = [format('%016x', digest.span_id)].pack('H16')
```

Comment:
Fix 2: Ensure otel spans are encoded in a manner that is consistent with the otel sdk: https://github.com/open-telemetry/opentelemetry-ruby/blob/5b2cc8ffe7614c27fd69f0b40d34368f0c06ff53/api/lib/opentelemetry/trace/span_context.rb#L42